python/cudf/cudf/pylibcudf_tests/conftest.py

```diff
@@ -29,3 +31,25 @@
 )
 def pa_type(request):
     return request.param
+
+
+# TODO: Test nullable data
+@pytest.fixture(scope="session")
+def pa_input_column(pa_type):
+    if pa.types.is_integer(pa_type) or pa.types.is_floating(pa_type):
+        return pa.array([1, 2, 3], type=pa_type)
+    elif pa.types.is_string(pa_type):
+        return pa.array(["a", "b", "c"], type=pa_type)
+    elif pa.types.is_boolean(pa_type):
+        return pa.array([True, True, False], type=pa_type)
+    elif pa.types.is_list(pa_type):
+        # TODO: Add heterogenous sizes
+        return pa.array([[1], [2], [3]], type=pa_type)
+    elif pa.types.is_struct(pa_type):
+        return pa.array([{"v": 1}, {"v": 2}, {"v": 3}], type=pa_type)
+    raise ValueError("Unsupported type")
+
+
+@pytest.fixture(scope="session")
+def input_column(pa_input_column):
+    return plc.interop.from_arrow(pa_input_column)
```

Comment:
I'm somewhat leery of using session-scoped fixtures for columnar data, mostly because I worry that writers of new tests won't know to look for these columns and the data might not generalize well across different classes of functions in different pylibcudf modules. These are currently just nebulous fears though. @bdice @brandon-b-miller what do you two think? If we want to use session-scope fixtures we should update the dev guide, IIRC I wrote in a recommendation for module-scope.

Reply:
I see your point here. There’s maybe an argument that the very simplest data fixtures (a non nullable numeric column for each type) could be reused, but the pattern you’re suggesting seems sensible too. I will move this.

Reply:
Generally speaking one would have a session (or module-scoped) fixture for things that are expensive to set up, or when one needs to share state between tests (urgh, no! :wink:). TBH, I would prefer to move these columns to their test use point and define the fixture in the (presumed) hashing test file. Rationale:

1. I always forget that conftest can define fixtures anyway, so I never look there
2. These names hog the namespace across all fixtures everywhere
3. As @vyasr says, they encode somewhat specific information (unlike the `pa_type` fixture which really is just "these are the types we are testing")